tests/samplers/test_batch.py

```diff
@@ -144,6 +145,20 @@ def test_weighted_sampling(self) -> None:
             for bbox in batch:
                 assert bbox == BoundingBox(0, 10, 0, 10, 0, 10)
 
+    def test_random_seed(self) -> None:
+        ds = CustomGeoDataset()
+        ds.index.insert(0, (0, 10, 0, 10, 0, 10))
+        sampler = RandomBatchGeoSampler(ds, 1, 1, generator=torch.manual_seed(0))
+        for bbox in sampler:
+            sample1 = bbox
+            break
+
+        sampler = RandomBatchGeoSampler(ds, 1, 1, generator=torch.manual_seed(0))
+        for bbox in sampler:
+            sample2 = bbox
+            break
```

Comment:
```suggestion
        sampler1 = RandomBatchGeoSampler(ds, 1, 1, generator=torch.manual_seed(0))
        sampler2 = RandomBatchGeoSampler(ds, 1, 1, generator=torch.manual_seed(0))
        sample1 = next(iter(sampler1))
        sample2 = next(iter(sampler2))
```
Same for other tests